.github/workflows/ci.yml

```diff
@@ -0,0 +1,56 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "main"
+
+jobs:
+  test:
+    name: Run tests & display coverage
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ 3.8, 3.9, "3.10", "3.11" ]
+    permissions:
+      pull-requests: write
+      contents: write
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+         fetch-depth: 1
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2.1.4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Poetry
+        uses: abatilo/actions-poetry@v2.1.6
+        with:
+          poetry-version: 1.6.1
+
+      - name: Set Poetry config
+        run: |
+          poetry config virtualenvs.path ~/.virtualenvs3.11
+
+      - name: Install dependencies
+        run: poetry install
```

Comment:
I think this will need to be `poetry install -E minify` for the minify tests to work correctly.

Reply:
yes, i just added that